### PR TITLE
NPM packaging tests

### PR DESCRIPTION
**Pull request by QnnOkabayashi**
This PR allows for using the code in `example/js` from a separate NPM package. To preface, we used to have `example/js/`, which contained some bindings and some ava tests. This PR changes that so we now have `example/js/{lib,app}`. All the old content now lives in `lib/`, and my "external" npm package that uses it lives in `app/`. 

I wrote all the changes at once and then tried to add changes in separate commits, but here's what each commit does respectively:

### Re-export from `api/`
Add `.d.ts` and `.js` files to re-export the contents of `lib/api/` into `lib/`. This is so we can use Rust to generate `lib/api/index.{js,d.ts}` files, and then have `lib/index.{js,d.ts}` files that never change and just re-export all contents from `lib/api/index.{js,d.ts}`. These exports under `lib/` allow us to call from outside libraries. I'm sure there's a better way to do this, but it didn't seem too hacky so I went with it.
### copy wasm into pkg
In order for external packages to be able to use what's in `lib/`, we have to reference the `.wasm` file relatively. I added changes so we now do that. This includes changing `gen_all.sh` to copying the compiled `.wasm` file from the `target` directory into `lib/`.
### Add `example/js/app/`
Added an "external" npm package that relies on `lib/`. This updates `ci.yml` to run `gen_all.sh` on the tests again to copy the `.wasm` files, and also runs the ava tests I put in `app/` (which are identical to the ones in `lib/`, except for where the imports come from).
### mv example/js/* example/js/lib/ _and_ change diplomat runtime from `.mjs` to `.js` 
Moved files into new `lib/` folder and changed `diplomat-runtime.mjs` to `diplomat-runtime.js` for reasons? idk it made it work.
### insta
update snapshots

### CODE REVIEW

**Comment by shadaj** [comment]
I'm (pleasantly?) surprised that we can switch from `.mjs` to `.js`! The original reason we needed that is that Node doesn't support ES modules out of the box and would force us to use `require`s otherwise. But perhaps something has changed in a recent Node release.

**Comment by Manishearth** [comment]
@shadaj I think we actually _had_ to switch to `.js` here, but also yeah I think you can tell npm about the kind of module in package.json now

File: example/js/app/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (sffc) [inline example/js/app/package.json:1]: Nit: Whenever there are any files that are generated by `tsc`, you should add the corresponding command line to a build script in package.json. See here for example:

https://docs.microsoft.com/en-us/visualstudio/javascript/compile-typescript-code-npm?view=vs-2022

File: example/js/lib/api/ICU4XFixedDecimalGroupingStrategy.js [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 };
+
 export const ICU4XFixedDecimalGroupingStrategy_rust_to_js = {
   0: "Auto",
   1: "Never",
   2: "Always",
   3: "Min2",
 };
+
+export const ICU4XFixedDecimalGroupingStrategy = {
```
Comment (sffc) [inline example/js/lib/api/ICU4XFixedDecimalGroupingStrategy.js:18]: Question: Why did this change? Stale diplomat output?

File: feature_tests/js/api/diplomat-runtime.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+function readString(wasm, ptr, len) {
```
Comment (sffc) [inline feature_tests/js/api/diplomat-runtime.js:1]: Issue: I think you didn't delete the .mjs version of this file as it looks like you intended to do

File: example/js/lib/api/ICU4XFixedDecimalGroupingStrategy.js (hunk 2 above)
Comment (Manishearth) [reply example/js/lib/api/ICU4XFixedDecimalGroupingStrategy.js:18]: yeah switching to cargo-make will improve the status quo here

File: example/js/app/package.json (hunk 1 above)
Comment (QnnOkabayashi) [reply example/js/app/package.json:1]: What corresponding command line? Do you mean `"build": "tsc"` under scripts?

File: example/js/lib/api/ICU4XFixedDecimalGroupingStrategy.js (hunk 2 above)
Comment (QnnOkabayashi) [reply example/js/lib/api/ICU4XFixedDecimalGroupingStrategy.js:18]: I added that in this PR. The reason for it was that the `.d.ts` file exported an enum type, and so the js side had to export some equivalent type. Maybe there's some other workaround? Not sure, but I think this is one of the reasons why unions of strings are preferred in TS over string enums,

**Review by sffc — APPROVED** [review]

**Pull request merged**